### PR TITLE
refactor(react-chart): move domains Getter to series component

### PR DESCRIPTION
**Pull request by DmitryBogomolov**
The **domains** *Getter* is moved from `ChartCore` to series component.
The behavior is changed as well. Previously all domains were built at once. Now domains are updated every time a series is added.

### CODE REVIEW

File: packages/dx-chart-core/src/plugins/scale/computeds.ts [hunk 1]
```diff
@@ -3,125 +3,122 @@ import { scaleLinear as d3ScaleLinear, scaleBand as d3ScaleBand } from 'd3-scale
...
-export const computeDomains: ComputeDomainsFn = (domains, seriesList) => {
-  const result = copy(domains);
-  collectDomainsFromSeries(result, seriesList);
-  calculateDomains(result, seriesList);
-  customizeDomains(result);
-  return result;
+export const extendDomains: ExtendDomainsFn = (domains, series) => {
+  const argumentDomain = updateDomain(
+    domains[ARGUMENT_DOMAIN], series, getArgument, getArgumentDomainItems);
+  const valueDomainName = getSeriesValueDomainName(series);
```
Comment (Krijovnick) [inline packages/dx-chart-core/src/plugins/scale/computeds.ts]: Why is `getValueDomainName` not called directly here?

File: packages/dx-chart-core/src/plugins/scale/computeds.ts [hunk 2]
```diff
@@ -3,125 +3,122 @@ import { scaleLinear as d3ScaleLinear, scaleBand as d3ScaleBand } from 'd3-scale
...
+  const valueDomain = updateDomain(
```
Comment (Krijovnick) [inline packages/dx-chart-core/src/plugins/scale/computeds.ts]: Why is the `updateDomainItems` method not called directly here?

File: packages/dx-chart-core/src/plugins/scale/computeds.ts (hunk 1 above)
Comment (DmitryBogomolov) [reply packages/dx-chart-core/src/plugins/scale/computeds.ts]: `getSeriesValueDomainName` is removed.

File: packages/dx-chart-core/src/plugins/scale/computeds.ts (hunk 2 above)
Comment (DmitryBogomolov) [reply packages/dx-chart-core/src/plugins/scale/computeds.ts]: `updateDomain` encloses and orders smaller domains updates (*factory* -> *items*) and is applied similarly to argument and value domains.

**Pull request merged**